### PR TITLE
Upgrade test - do not rerun green tests

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- all green test won't be retried
- the state is shared by config map

**Related issue(s)**

### CODE REVIEW

File: tests/end-to-end/upgrade/chart/upgrade/templates/tests-info.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-{{ template "fullname" . }}-info
+data: {}
```
Comment (polskikiel) [inline tests/end-to-end/upgrade/chart/upgrade/templates/tests-info.yaml:5]: minor: you don't have to specify data

File: tests/end-to-end/upgrade/README.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ Use the following environment variables to configure the application:
 | **APP_KUBECONFIG_PATH** | NO | None | A path to the `kubeconfig` file needed to run an application outside of the cluster. |
 | **APP_MAX_CONCURRENCY_LEVEL** | NO | `1` | A maximum concurrency level used for running tests. |
 | **APP_TESTING_ADDONS_URL** | YES | None | An external link to testing addons. |
+| **APP_WORKING_NAMESPACE** | No | `e2e-upgrade-test` | A namespace where the test is run. |
```
Comment (klaudiagrz) [inline tests/end-to-end/upgrade/README.md:38]:
```suggestion
| **APP_WORKING_NAMESPACE** | No | `e2e-upgrade-test` | A Namespace in which the test is run. |
```

File: tests/end-to-end/upgrade/README.md [hunk 3]
```diff
@@ -35,6 +35,8 @@ Use the following environment variables to configure the application:
...
+| **APP_TESTS_INFO_CONFIG_MAP_NAME** | No | `upgrade-tests-info` | A name of a config map, where the upgrade test framework stores information about passed tests. |
```
Comment (klaudiagrz) [inline tests/end-to-end/upgrade/README.md:39]:
```suggestion
| **APP_TESTS_INFO_CONFIG_MAP_NAME** | No | `upgrade-tests-info` | A name of a ConfigMap in which the upgrade test framework stores information about passed tests. |
```

**Pull request merged**